crates/starknet-devnet/tests/test_dump_and_load.rs

```diff
@@ -100,6 +100,49 @@ mod dump_and_load_tests {
         assert!(devnet_dump.is_err());
     }
 
+    #[tokio::test]
+    async fn dump_and_load_blocks_on_demand() {
+        let modes = vec!["exit", "block"];
+
+        for mode in modes {
+            let dump_file = UniqueAutoDeletableFile::new(
+                ("dump_load_dump_load_on_".to_owned() + mode).as_str(),
+            );
+
+            for _ in 0..2 {
+                let devnet_dump = BackgroundDevnet::spawn_with_additional_args(&[
+                    "--dump-path",
+                    &dump_file.path,
+                    "--dump-on",
+                    mode,
+                    "--blocks-on-demand",
+                ])
+                .await
+                .expect("Could not start Devnet");
+
+                devnet_dump.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+                devnet_dump.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+                devnet_dump.create_block().await.unwrap();
+
+                send_ctrl_c_signal_and_wait(&devnet_dump.process).await;
+            }
+
+            let devnet_load = BackgroundDevnet::spawn_with_additional_args(&[
+                "--dump-path",
+                &dump_file.path,
+                "--dump-on",
+                mode,
+                "--blocks-on-demand",
+            ])
+            .await
+            .expect("Could not start Devnet");
+
+            let last_block = devnet_load.get_latest_block_with_tx_hashes().await.unwrap();
+            assert_eq!(last_block.block_number, 2);
+            assert_eq!(last_block.transactions.len(), 2);
```

Comment:
But shouldn't it be 1? 0 is genesis block and 1 is the only new created block (containing two minting txs)?

Reply:
There is loop with `for _ in 0..2`.

This loop was needed to test all scenarios. As far as I remember it was related to doublets.

Reply:
I completely forgot it's inside the loop, it's correct then. Just, in that case, I'd suggest extracting the `2` to a variable (`total_iterations` or something like that) which is used both in loop definition and in assertions.

Reply:
fixed in 9a96b6e9126187cc44c2d4e2f70163dfb494a8b3